src/services/transfer/transfer-out-fragments.js

```diff
@@ -1,26 +1,101 @@
+    let count = 0;
+
+    for (const messageIdWithReplacement of messageIdsWithReplacements) {
+      try {
+        const {oldMessageId, newMessageId} = messageIdWithReplacement;
+        let fragment = await getFragment(conversationIdFromEhrIn, oldMessageId);
+        fragment = replaceMessageIdsInObject(fragment, messageIdsWithReplacements);
 
-  logInfo(`All fragments have been successfully sent to GP2GP Messenger, Inbound Conversation ID: ${conversationId}`);
-}
\ No newline at end of file
+        await sendFragment(conversationId, odsCode, fragment, newMessageId);
+
+        logInfo(`Fragment ${++count} of ${messageIdsWithReplacements.length} sent to the GP2GP Messenger - with old Message ID ${oldMessageId}, and new Message ID ${newMessageId}.`);
+
+        if (config.fragmentTransferRateLimitTimeoutMilliseconds)
+          await new Promise(executor => setTimeout(executor, config.fragmentTransferRateLimitTimeoutMilliseconds));
+      } catch (error) {
+        await handleFragmentTransferError(error, conversationId, messageIdWithReplacement.newMessageId);
+        throw error;
+      }
+    }
+    logInfo(`All fragments have been successfully sent to GP2GP Messenger.`);
+}
+
+const getMessageIdsOfFragmentsEligibleForSending = async (messageIdsWithReplacements) => {
+  const newMessageIds = messageIdsWithReplacements.map(messageIdWithReplacement => messageIdWithReplacement.newMessageId);
+  const messageFragmentRecords = await getAllMessageFragmentRecordsByMessageIds(newMessageIds);
+
+  const messageIdsOfFragmentsEligibleForSending = newMessageIds.filter(messageId => {
+    const messageFragmentRecord = messageFragmentRecords.find(
+      messageFragmentRecord => messageFragmentRecord.messageId === messageId);
+
+    return isFragmentIsEligibleToBeSent(messageFragmentRecord);
+  });
+
+  logInfo(`Out of ${newMessageIds.length} message Ids, ` +
+  `${newMessageIds.length - messageIdsOfFragmentsEligibleForSending.length} have already been sent. ` +
+  `${messageIdsOfFragmentsEligibleForSending.length} are eligible to be sent`);
+
+  return messageIdsOfFragmentsEligibleForSending;
+}
+
+const isFragmentIsEligibleToBeSent = (messageFragmentRecord) => {
+  // if the fragment has no fragmentRecord, it is considered eligible for sending
+  const fragmentStatus = messageFragmentRecord?.status;
+  return !(fragmentStatus === Status.SENT_FRAGMENT || fragmentStatus === Status.MISSING_FROM_REPO);
+}
+
+const handleFragmentTransferError = async (error, conversationId, messageId) => {
+  switch (true) {
+    case error instanceof PresignedUrlNotFoundError:
+      await updateFragmentStatus(conversationId, messageId, Status.MISSING_FROM_REPO);
+      break;
+    case error instanceof DownloadError:
+      await updateFragmentStatus(conversationId, messageId, Status.DOWNLOAD_FAILED);
+      break;
+    // this will catch any miscellaneous errors

```

Comment:
Redundant comment - can be removed, nature of `default` is to catch any unmatching criteria.